### PR TITLE
create what's new for April

### PR DESCRIPTION
**Pull request by BillWagner**
Run the app to create "what's new" for April.

### CODE REVIEW

File: docs/whats-new/dotnet-2020-04.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+### Updated articles
+
+- [App configuration](../architecture/blazor-for-web-forms-developers/config.md) - Blazor ebook config
+
+## .NET Core
+
+### New articles
+
+- [Trim self-contained deployments and executables](../core/deploying/trim-self-contained.md) - Create trimming doc
+- [Ubuntu 20.04 Package Manager - Install .NET Core](../core/install/linux-package-manager-ubuntu-2004.md) - Install for Ubuntu 20.04
```
Comment (pkulikov) [inline docs/whats-new/dotnet-2020-04.md:26]: Given #18092, should it be here?
/cc @Thraka

File: docs/whats-new/dotnet-2020-04.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+### New articles
+
+- [Reserved attributes: Determine caller information](../csharp/language-reference/attributes/caller-information.md) - Reorganize and update language reference for attributes processed by the C# compiler, including nullable analysis
+- [Reserved attributes: ConditionalAttribute, ObsoleteAttribute, AttributeUsageAttribute](../csharp/language-reference/attributes/general.md) - Reorganize and update language reference for attributes processed by the C# compiler, including nullable analysis
+- [Reserved attributes: Assembly level attributes](../csharp/language-reference/attributes/global.md) - Reorganize and update language reference for attributes processed by the C# compiler, including nullable analysis
+- [Reserved attributes contribute to the compiler's null state static analysis](../csharp/language-reference/attributes/nullable-analysis.md) - Reorganize and update language reference for attributes processed by the C# compiler, including nullable analysis
+- [Nullable reference types (C# reference)](../csharp/language-reference/builtin-types/nullable-reference-types.md) - Update language reference for nullable token and related topics
+
+### Updated articles
```
Comment (pkulikov) [inline docs/whats-new/dotnet-2020-04.md:38]: The [Structure types](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/struct) article has got a lot of updates in March and in April. I agree those have been mostly consolidations, not new material. Only wonder if the omission of that article is be design or by accident.

Comment (BillWagner) [reply docs/whats-new/dotnet-2020-04.md:38]: That's by accident. I'll add it. Thanks for pointing this out.

File: docs/whats-new/dotnet-2020-04.md (hunk 1 above)
Comment (Thraka) [reply docs/whats-new/dotnet-2020-04.md:26]: Yeah remove that. Also, with creating the PR for review prior to the sprint end but accepted today (#18080)[https://github.com/dotnet/docs/pull/18080] Does that appear next sprint whats new or this one?

**Review by gewarren — APPROVED** [review]
LG. A few comments:

- There will be some updates to https://docs.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props worth mentioning (does the tool go off PRs in the April project?).
- Should we add redirects for the deleted pages each month?
- I also added some breaking changes for .NET Core 1.0 and 3.0 and .NET 5 (Scott Addie did too).

**Comment by BillWagner** [comment]
@gewarren @scottaddie 

> - There will be some updates to https://docs.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props worth mentioning (does the tool go off PRs in the April project?). 
The code goes off the project / sprint dates. So, those should show up next month.
> - Should we add redirects for the deleted pages each month?
So far, we've decided against it. This is another use case for folder level redirects, where any outdated link could redirect to the "what's new" landing page.  As it exists now, this becomes another tax on the global redirection file.
> - I also added some breaking changes for .NET Core 1.0 and 3.0 and .NET 5 (Scott Addie did too).
Can you and Scott make PRs to add those by hand? Then, they are highlighted, and I can figure out why the tool missed them and fix it.

**Comment by gewarren** [comment]
> Can you and Scott make PRs to add those by hand? Then, they are highlighted, and I can figure out why the tool missed them and fix it.

If What's new for April is for work completed in March, then these should just show up next month.

**Pull request merged**